src/Features/Core/Portable/ExternalAccess/Copilot/Api/ICopilotMapCodeService.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Copilot.Api
+{
+    internal interface IMapCodeService : ILanguageService
+    {
+        Task<Document> MapCodeAsync(
+            Document document,
+            ImmutableArray<string> contents,
+            ImmutableArray<DocumentSpan> focusLocations,
+            bool formatMappedCode,
+            CancellationToken cancellationToken);
+    }
```

Comment:
so the idea is that they will implement this interface?

Reply:
Yes. And we will implemented the LSP handler for a `mapCode` request, which will try to use MEF imported service